helpdesk_mgmt/models/helpdesk_ticket.py

```diff
@@ -0,0 +1,246 @@
+        ticket = self[0]
+        if "stage_id" in tracking and ticket.stage_id.mail_template_id:
+            res["stage_id"] = (
+                ticket.stage_id.mail_template_id,
+                {
+                    "auto_delete_message": True,
+                    "subtype_id": self.env["ir.model.data"].xmlid_to_res_id(
+                        "mail.mt_note"
+                    ),
+                    "email_layout_xmlid": "mail.mail_notification_light",
+                },
+            )
+        return res
+
+    @api.model
+    def message_new(self, msg, custom_values=None):
+        """ Override message_new from mail gateway so we can set correct
+        default values.
+        """
+        if custom_values is None:
+            custom_values = {}
+        defaults = {
+            "name": msg.get("subject") or _("No Subject"),
+            "description": msg.get("body"),
+            "partner_email": msg.get("from"),
+            "partner_id": msg.get("author_id"),
+        }
+        defaults.update(custom_values)
+
+        # Write default values coming from msg
+        ticket = super().message_new(msg, custom_values=defaults)
+
+        # Use mail gateway tools to search for partners to subscribe
+        email_list = tools.email_split(
+            (msg.get("to") or "") + "," + (msg.get("cc") or "")
+        )
+        partner_ids = [
+            p
+            for p in self.env["mail.thread"]._mail_find_partner_from_emails(
+                email_list, records=ticket, force_create=False
+            )
+            if p
+        ]
+        ticket.message_subscribe(partner_ids)
+
+        return ticket
+
+    def message_update(self, msg, update_vals=None):
+        """ Override message_update to subscribe partners """
+        email_list = tools.email_split(
+            (msg.get("to") or "") + "," + (msg.get("cc") or "")
+        )
+        partner_ids = [
+            p
+            for p in self.env["mail.thread"]._mail_find_partner_from_emails(
+                email_list, records=self, force_create=False
+            )
+            if p
+        ]
+        self.message_subscribe(partner_ids)
+        return super().message_update(msg, update_vals=update_vals)
+
+    def message_get_suggested_recipients(self):
+        recipients = super().message_get_suggested_recipients()
+
+        for ticket in self:
+            reason = (
+                _("Partner Email")
+                if ticket.partner_id and ticket.partner_id.email
+                else _("Partner Id")
+            )
+
+            if ticket.partner_id and ticket.partner_id.email:
+                ticket._message_add_suggested_recipient(
+                    recipients, partner=ticket.partner_id, reason=reason
+                )
+            elif ticket.partner_email:
+                ticket._message_add_suggested_recipient(
+                    recipients, email=ticket.partner_email, reason=reason
+                )
+        return recipients
```

Comment:
```suggestion
    def _message_get_suggested_recipients(self):
        recipients = super()._message_get_suggested_recipients()
        try:
            for ticket in self:
                if ticket.partner_id:
                    ticket._message_add_suggested_recipient(
                        recipients,
                        partner=ticket.partner_id,
                        reason=_('Customer'))
                elif ticket.partner_email:
                    ticket._message_add_suggested_recipient(
                        recipients,
                        email=ticket.partner_email,
                        reason=_('Customer Email'))
        except AccessError:
            pass
        return recipients
```

For send message to suggested recipients like in crm.

Reply:
It's not done, is '_message_get_suggested_recipients' not 'message_get_suggested_recipients'.

Reply:
@carlos-domatix yeah typo.